operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -420,295 +476,406 @@ private static Uri uri(FHIROperationContext operationContext, String uriPath) {
+                (allowedIncludes == null || allowedIncludes.contains(parameterName)
+                || allowedIncludes.contains(simplifiedParameterName)) &&
+                searchParam != null) {
             searchParameters.add("_include", parameterName);
         } else {
             if (LOG.isLoggable(Level.FINE)) {
                 LOG.fine("Filtering out searchParameter because it is not supported by the server config: " + parameterName);
             }
         }
     }
+
+    /**
+     * Read additional associated resources for the "extra" types (like location)
+     * supported by this server configuration.  Only LITERAL_RELATIVE references are
+     * resolved as LITERAL_ABSOLUTE references that match the server's hostname are
+     * normalized prior to this method.
+     *
+     * @param compartmentMemberType the type of resource currently drilling down on
+     * @param newEntries a list of Bundle response entries that include compartmentMemberType instances found via search
+     * @param allEntries a list of all entries that have been found so far (minus duplicates)
+     * @param resourceHelper an instance of FHIRResourceHelpers to use to make the calls
+     * @param extraResources a list of "extra" resource types configured for this server
+     */
+    private void readsOfAdditionalAssociatedResources(String compartmentMemberType, List<Entry> newEntries,
+                List<Entry> allEntries, List<String> resourceIds,
+                FHIRResourceHelpers resourceHelper, List<String> extraResources) throws Exception {
+        if (extraResources == null || extraResources.isEmpty()) {
+            return;
+        }
+        Bundle requestBundle = Bundle.builder().type(BundleType.BATCH).build();
+        String url = ReferenceUtil.getServiceBaseUrl();
+
+        for (Entry entry: newEntries) {
+            // Look up entries by reference
+            List<Reference> references = ReferenceFinder.getReferences(entry.getResource());
+            for (Reference reference: references) {
+                ReferenceValue referenceValue = ReferenceUtil.createReferenceValueFrom(reference, url);
+                if (referenceValue.getType().equals(ReferenceValue.ReferenceType.LITERAL_RELATIVE)) {
+                    String type = referenceValue.getTargetResourceType();
+                    String value = referenceValue.getValue();
+                    String resourceId = type + "/" + value;
+                    if (!resourceIds.contains(resourceId) && extraResources.contains(type) && (type.equals(ResourceType.Value.LOCATION.value()) || type.equals(ResourceType.Value.MEDICATION.value())
+                            || type.equals(ResourceType.Value.ORGANIZATION.value()) || type.equals(ResourceType.Value.PRACTITIONER.value()))) {
+                        // Bundle up the requests so we can send them as a batch
+                        requestBundle = addRequestToBundle(requestBundle, HTTPVerb.GET, resourceId, null);
```

Comment:
it would be slightly better to build up a list incrementally and then just build the Bundle once at the end.
that way you don't pay the cost of copying the bundle (`bundle.toBuilder()`) on each iteration.

if you want you could keep requestBundleBuilder as a var and pass that to addRequestToBundle instead of the built bundle

Reply:
line 803 would become:
```
Bundle.Builder requestBundleBuilder = Bundle.builder().type(BundleType.BATCH);
```

and `addRequestToBundle` would take and return a Bundle.Builder.

Reply:
Done